packages/babel-helpers/src/helpers/usingCtx.ts

```diff
@@ -30,18 +45,24 @@ export default function _usingCtx() {
       // core-js-pure uses Symbol.for for polyfilling well-known symbols
       if (isAwait) {
         var dispose =
-          value[Symbol.asyncDispose || Symbol.for("Symbol.asyncDispose")];
+            value[Symbol.asyncDispose || Symbol.for("Symbol.asyncDispose")],
```

Comment:
If `Symbol.asyncDispose || Symbol.for("Symbol.asyncDispose")` is `undefined`, which implies that the dispose symbol is not properly polyfilled, we can provide a better error such as "`Symbol.asyncDispose` is not defined" rather than throwing "object is not disposable" as we are reading the `"undefined"` property of the `value`.

Reply:
I prefer to unify them into something like "disposal function is undefined". :)

Reply:
The disposal function may be okay. The proposed error here is for the case when `Symbol.asyncDispose` is not defined, rather than `resource[Symbol.asyncDispose]` is not defined.

Reply:
You are right! I changed it to `invalid`.

Reply:
I just realized that `Symbol.asyncDispose || Symbol.for("Symbol.asyncDispose")` will always return a symbol. If `Symbol.dispose` is not polyfilled, user will end up defining a disposal function under the name `undefined` without any runtime error. So if we can't find the disposal function and `value.undefined` is a function, this suggests that the user may not polyfill `Symbol.dispose` and in this case we can provide a better error message. Anyway it will be addressed in another PR.

I think we can add a section to the plugin docs about how to use this plugin with core-js 3 or other polyfill providers. Integration tests with polyfill providers should be added as well.

Reply:
`Symbol.asyncDispose || Symbol.for("Symbol.asyncDispose")` always returns a symbol, so shouldn't we always look for `value[symbol]`?

Reply:
In chrome 122 or any browsers without `@@dispose` support,
```js
const disposable = { [Symbol.dispose]() {} }
```
will be evaluated as if it were defined as
```js
const disposable = { undefined() {} }
```

If users have properly polyfilled `Symbol.dispose`, the transformed result will be evaluated as if
```js
const disposable = { [Symbol.for("Symbol.dispose")]() {} }
```

The `usingCtx` helper work when 1) the browser has native support of `@@dispose` or 2) the `@@dispose` has been polyfilled correctly. However, in the case when a polyfill is missing, the helper will throw `object is not disposable` because the desired disposal function is defined at the `undefined` property, which can be very confusing for end users because they thought they have defined it, without realizing that the symbol is not polyfilled. In this case we should suggest users that `@@dispose` may not be properly polyfilled.

Reply:
Makes sense! But given that this exists in all symbol-related helpers, I'm a little hesitant that we should include it by default.